git/foundations_git/commit_messages.md

```diff
@@ -20,14 +20,47 @@ Yes! Let me give you a quick list of reasons why:
 
 -   Having a good commit message history will also be helpful if you come back to a project you were working on after stepping away from it for a while. You likely won't remember your thought process and changes made when initially writing out the code. 
 
+### Bad vs. Good Commits
+
+When it comes to writing commits, it is crucial to know how to write them effectively. Here's an example of a bad commit message:
+
+`fix a bug`
+
+Even though it describes what you did, the message is too vague, which leaves the other developers on your team confused. A good commit message will explain the why behind your changes. In other words, a commit message describes what problem your changes solve and how it solves them. 
+
+Effective commits consist of two separate parts: a subject, and a body:
+
+## Subject
+A brief summary of the change you made to the project.  **Note:** GitHub has a 50-character limit so we recommend keeping your commits' subject to within this amount. 
+
+`This is the change I made to the codebase.`
+
+## Body
+A concise yet clear description of what you did.  
+
+`Describe the problem that your changes solve and how in 50 words or less`
```

Comment:
I agree with putting a constraint on the body length and don't think it's a bad move to include it. But theres no hard and fast rule with this unfortunately. 

I'd rather we omit putting a hard constraint on this to avoid us getting into an endless cycle of future PR's where people disagree with our chosen limit and suggest increasing or decreasing the limit - that kind of thing happens way too often lol. 

Also a code block for this too so we're consistent in the formatting:

```suggestion
~~~
Describe the problem your commit solves and how.
~~~
```